### PR TITLE
Fix some DNN samples for compatibility with Python 3

### PR DESCRIPTION
**Pull request by catree**
### This pullrequest introduces
small changes to be able to run some DNN samples with Python 3:

- integer division returns float in Python 3, [see](https://stackoverflow.com/questions/1282945/python-integer-division-yields-float)
- long renamed to int in Python 3

I hope these changes do not break the compatibility with Python 2.

### CODE REVIEW

File: samples/dnn/colorization.py [hunk 1]
```diff
@@ -27,8 +29,8 @@ def parse_args():
 
     # populate cluster centers as 1x1 convolution kernel
     pts_in_hull = pts_in_hull.transpose().reshape(2, 313, 1, 1)
-    net.getLayer(long(net.getLayerId('class8_ab'))).blobs = [pts_in_hull.astype(np.float32)]
```
Comment (dkurt) [inline samples/dnn/colorization.py]: @catree, It'll break the script at https://github.com/opencv/opencv/blob/master/modules/dnn/misc/python/pyopencv_dnn.hpp#L9. Could you check it and  modify the function to be more flexible?

File: samples/dnn/mobilenet_ssd_python.py [hunk 2]
```diff
@@ -99,7 +99,7 @@
         y2 = y1 + cropSize[1]
         x1 = (cols - cropSize[0]) / 2
         x2 = x1 + cropSize[0]
-        frame = frame[y1:y2, x1:x2]
+        frame = frame[int(y1):int(y2), int(x1):int(x2)] #for Python 3 compatibility (2/2=1.0)
```
Comment (dkurt) [inline samples/dnn/mobilenet_ssd_python.py]: I think `y1 = int((rows - cropSize[1]) / 2)`, `x1 = int((cols - cropSize[0]) / 2)` is even better.

File: samples/dnn/colorization.py (hunk 1 above)
Comment (catree) [reply samples/dnn/colorization.py]: It works now for both Python version.

Comment (dkurt) [reply samples/dnn/colorization.py]: @catree, So, we can emit the casting here?

**Comment by dkurt** [comment]
:+1: It works for me! Thank you!

**Pull request merged**